app/js/services/actions.service.js

```diff
@@ -429,22 +429,10 @@ function ActionsService(
         let systemDeferred;
 
         // if we already have all of the systems affected use vars.allSystems
-        if (vars.allSystems === null) {
```

Comment:
No need to take out the caching. Really we just need to verify that if vars.allSystems exist that the count of allSystems is equal to vars.totalRuleSystems. In other words:

`if (vars.allSystems === null || vars.allSystems.length === vars.totalRuleSystems) {`